File: GestionEmpleadosFrontend/src/app/models/departamento.model.ts
export interface Departamento {
  departamento_id: number;
  nombre: string;
  ubicacion: string;
  jefe_departamento?: string;
  extension?: string;
}

export interface DepartamentoCreate {
  nombre: string;
  ubicacion: string;
  jefe_departamento?: string;
  extension?: string;
}

export interface DepartamentoUpdate {
  departamento_id: number;
  nombre: string;
  ubicacion: string;
  jefe_departamento?: string;
  extension?: string;
}

File: GestionEmpleadosBackend/Models/DepartamentoModel.cs
using System.ComponentModel.DataAnnotations;

namespace GestionEmpleados.Models
{
    public class DepartamentoModel
    {
        [Key]
        public int departamento_id { get; set; }

        [Required]
        [StringLength(100)]
        public string nombre { get; set; }

        [Required]
        [StringLength(200)]
        public string ubicacion { get; set; }

        [StringLength(100)]
        public string jefe_departamento { get; set; }

        [StringLength(20)]
        public string extension { get; set; }

        // Relación con asignaciones
        public virtual ICollection<AsignacionModel> Asignaciones { get; set; } = new List<AsignacionModel>();
    }
}